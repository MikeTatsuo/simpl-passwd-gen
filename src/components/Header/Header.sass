.header
	display: flex
	grid-area: header

	&-title
		align-self: center
		justify-self: center
		flex-grow: 1
		text-align: center
