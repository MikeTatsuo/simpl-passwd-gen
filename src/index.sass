@import styles/utils
@import styles/base
@import styles/themes
@import styles/layout
