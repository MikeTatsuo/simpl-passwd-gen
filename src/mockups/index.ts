export * from './Checkbox'
export * from './Header';
