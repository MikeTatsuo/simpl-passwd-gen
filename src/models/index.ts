export * from './CardFactoryTypes.enum';
export * from './CardProps.interface';
export * from './CheckboxProps.interface';
export * from './HeaderProps.interface';
