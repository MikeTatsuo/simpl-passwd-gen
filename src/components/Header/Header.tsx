import { Component } from 'react';

import { HeaderProps } from 'src/models';

import './Header.sass'

export default class Header extends Component<HeaderProps> {
	render() {
		return (
			<header className="header">
				<span className="header-title">{this.props.title ?? ''}</span>
			</header>
		);
	}
}
