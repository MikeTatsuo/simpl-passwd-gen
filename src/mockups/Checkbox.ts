import { datatype, lorem } from 'faker';

import { CheckboxProps } from "src/models";

export const CheckboxMockup: CheckboxProps = {
	checked: datatype.boolean(),
	label: lorem.word(),
	name: lorem.word(),
};
