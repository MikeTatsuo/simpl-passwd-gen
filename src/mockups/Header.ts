import { lorem } from 'faker';

import { HeaderProps } from 'src/models';

export const HeaderMockup: HeaderProps = {
	title: lorem.sentence(),
}
