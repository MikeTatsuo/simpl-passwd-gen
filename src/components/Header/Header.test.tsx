import { unmountComponentAtNode, render } from 'react-dom';
import { act } from 'react-dom/test-utils';

import Header from './Header';

import { HeaderMockup } from 'src/mockups';

const { title } = HeaderMockup;

let container: HTMLDivElement;

describe('Header.tsx', () => {
	beforeEach(() => {
		container = document.createElement('div');
		document.body.appendChild(container);
	});

	afterEach(() => {
		unmountComponentAtNode(container);
		container.remove();
	});

	describe('snapshot', () => {
		it('should match snapshot', () => {
			act(() => {
				render(<Header />, container);
			});

			expect(container).toMatchSnapshot();
		});
	});

	describe('data', () => {
		it('renders header title', () => {
			act(() => {
				render(<Header title={title} />, container);
			});

			expect(container.textContent).toMatch(title as string);
		});
	});
});
