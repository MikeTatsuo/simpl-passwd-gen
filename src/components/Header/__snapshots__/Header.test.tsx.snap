// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header.tsx snapshot should match snapshot 1`] = `
<div>
  <header
    class="header"
  >
    <span
      class="header-title"
    />
  </header>
</div>
`;
